src/parser.rs

```diff
@@ -3467,10 +3467,9 @@ impl<'a> Parser<'a> {
                 Keyword::STRING => Ok(DataType::String),
                 Keyword::TEXT => Ok(DataType::Text),
                 Keyword::BYTEA => Ok(DataType::Bytea),
-                Keyword::NUMERIC | Keyword::DECIMAL | Keyword::DEC => {
-                    let (precision, scale) = self.parse_optional_precision_scale()?;
```

Comment:
> parse_optional_precision_scale

is this function still used anywhere? If not shall we remove it? If so, perhaps there are other places that should be migrated over?

Reply:
It is used in another place, in the parser as well. TBH I didn't review the logic for the other usage, maybe it ends in the same problem, but I'll take a look in another moment.

I think it can be left like that for now.